samcli/lib/build/app_builder.py

```diff
@@ -218,10 +218,12 @@ def update_template(template_dict: Dict, original_template_path: str, built_arti
             properties = resource.setdefault("Properties", {})
 
             if resource_type == SamBaseProvider.SERVERLESS_FUNCTION and properties.get("PackageType", ZIP) == ZIP:
-                properties["CodeUri"] = store_path
+                if "InlineCode" not in properties or properties["InlineCode"] is None:
+                    properties["CodeUri"] = store_path
 
             if resource_type == SamBaseProvider.LAMBDA_FUNCTION and properties.get("PackageType", ZIP) == ZIP:
-                properties["Code"] = store_path
+                if "ZipFile" not in properties["Code"] or properties["Code"]["ZipFile"] is None:
```

Comment:
We may have a condition where `Code` is not in `properties` (line 218), you may want to check it.

Reply:
makes sense, I will add the logic here

Reply:
Also we could maybe get the `PackageType` value into a variable instead of calling get with a default 4 times.


```python
package_type = properties.get("PackageType", ZIP)

if package_type == ZIP:
    if resource_type == SamBaseProvider.SERVERLESS_FUNCTION and ("InlineCode" not in properties or properties["InlineCode"] is None):
        ...

    if resource_type == SamBaseProvider.LAMBDA_FUNCTION and ("ZipFile" not in properties["Code"] or properties["Code"]["ZipFile"] is None):
         ...

if package_type == IMAGE:
     ...
```

Reply:
just saw that it is done this way in the other files, `packagetype` is without a `_` though.

Reply:
good suggestion!

Reply:
@mgrandis the "packagetype" you see in other files is an attribute name in provider.py. I think package_type would be the better name here

Reply:
I would go with `package_type`, always easier to read with underscores.